test/e2e/volumes.go

```diff
@@ -549,4 +633,77 @@ var _ = Describe("Volumes", func() {
 		})
 	})
 
+	////////////////////////////////////////////////////////////////////////
+	// OpenStack Cinder
+	////////////////////////////////////////////////////////////////////////
+
+	// Marked with [Skipped] to skip the test by default (see driver.go),
+	// The test assumes that OpenStack client tools are installed
+	// (/usr/bin/nova, /usr/bin/cinder and /usr/bin/keystone)
+	// and that the usual OpenStack authentication env. variables are set
+	// (OS_USERNAME, OS_PASSWORD, OS_TENANT_NAME at least).
+
+	Describe("[Skipped] Cinder", func() {
+		It("should be mountable", func() {
+			config := VolumeTestConfig{
+				namespace: namespace.Name,
+				prefix:    "cinder",
+			}
+
+			// We assume that namespace.Name is a random string
+			volumeName := namespace.Name
+			By("creating a test Cinder volume")
+			output, err := exec.Command("cinder", "create", "--display-name="+volumeName, "1").CombinedOutput()
+			outputString := string(output[:])
+			Logf("cinder output:\n%s", outputString)
+			Expect(err).NotTo(HaveOccurred())
+
+			defer func() {
+				// Ignore any cleanup errors, there is not much we can do about
+				// them. They were already logged.
+				deleteCinderVolume(volumeName)
+			}()
+
+			// Parse 'id'' from stdout. Expected format:
+			// |     attachments     |                  []                  |
+			// |  availability_zone  |                 nova                 |
+			// ...
+			// |          id         | 1d6ff08f-5d1c-41a4-ad72-4ef872cae685 |
+			volumeID := ""
+			for _, line := range strings.Split(outputString, "\n") {
+				fields := strings.Fields(line)
+				if len(fields) != 5 {
+					continue
+				}
+				if fields[1] != "id" {
+					continue
+				}
+				volumeID = fields[3]
+				break
+			}
+			Logf("Volume ID: %s", volumeID)
+			Expect(volumeID).NotTo(Equal(""))
+
+			defer func() {
+				if clean {
+					Logf("Running volumeTestCleanup")
+					volumeTestCleanup(c, config)
```

Comment:
I tried today and quite contrary to my assumption, second test failure is not logged at all:
```go
someTest() {
	defer func() {
		glog.Infof("cleanup here!")
		err := fmt.Errorf("This is the second error")
		expectNoError(err, "Got an error: %v", err)
	}()
	err := fmt.Errorf("This is the first error")
	expectNoError(err, "Got an error: %v", err)
}
```

In the log I see:
```
...
cleanup here!
....

    Got an error: This is the first error
    Expected error:
        <*errors.errorString | 0xc82045c3b0>: {
            s: "This is the first error",
        }
        This is the first error
    not to have occurred
```

I'm adding patch with some extra logs *inside* `volumeTestCleanup()`.